File: bench/py/71_exc_no_throw.py
# try/except that NEVER throws — the no-exception hot-path handler overhead.
import sys
scale = int(sys.argv[1]) if len(sys.argv) > 1 else 1

N = 500000 * scale
s = 0
caught = 0

for i in range(N):
    try:
        s = (s + i) % 1000000007
    except ZeroDivisionError:
        caught += 1                  # never taken

print("result:", s, caught)

File: bench/cpp/71_exc_no_throw.cpp
/* SPDX-License-Identifier: BSD-2-Clause */
/* Faithful C++ of bench/my/71_exc_no_throw.my: a try/catch that NEVER throws -
 * measures the pure handler-region overhead on the no-exception HOT PATH (with
 * zero-cost EH, ~free). `s` is a serial modular-sum dependency (no barrier
 * needed); `caught` stays 0. Both feed the printed result. */
#include "bench.h"

struct DivByZero {};

int main(int argc, char **argv)
{
    long scale = bench_scale(argc, argv);
    long N = 500000L * scale;
    long s = 0;
    long caught = 0;

    for (long i = 0; i < N; i++) {
        try {
            s = (s + i) % 1000000007;
        } catch (const DivByZero &) {
            caught++;                    /* never taken */
        }
    }

    printf("result: %ld %ld\n", s, caught);
    return 0;
}
